api/types/license.go

```diff
@@ -448,6 +547,18 @@ type LicenseSpecV3 struct {
 	SupportsMachineID Bool `json:"machine_id,omitempty"`
 	// SupportsResourceAccessRequests turns resource access request support on or off
 	SupportsResourceAccessRequests Bool `json:"resource_access_requests,omitempty"`
+	// SupportsDeviceTrust turns Device Trust on or off
+	SupportsDeviceTrust Bool `json:"device_trust,omitempty"`
+	// SupportsOIDC turns OIDC connectors on or off
+	SupportsOIDC Bool `json:"oidc"`
+	// SupportsSAML turns SAML connectors on or off
+	SupportsSAML Bool `json:"saml"`
+	// SupportsAccessControls turns FIPS access controls on or off
+	SupportsAccessControls Bool `json:"access_controls"`
+	// SupportsHSM turns hardware security modules on or off
+	SupportsHSM Bool `json:"hsm"`
 	// Trial is true for trial licenses
 	Trial Bool `json:"trial,omitempty"`
+	// FeatureSource is the source of the set of enabled feature
+	FeatureSource FeatureSource `json:"feature_source"`
```

Comment:
I find this a bit confusing. The struct is called `LicenseSpecV3`, but we now have a struct called "license" with a feature source that is something other than the license?

Reply:
That sounds cleaner!